lib/training/wiki_slide_parser.rb

```diff
@@ -216,9 +216,13 @@ def module_hash(template, module_number)
   end
 
   def template_parameter_value(template, parameter)
-    # Extract value from something like:
-    # | parameter_name = value
-    match = template.match(/\|\s*#{parameter}\s*=\s*(?<value>.*)/)
-    match && match['value']
+    # Extract value from the template
+    match = template.match(/\|\s*#{parameter}\s*=\s*(?<value>.*?)(?=\||\Z)/m)
```

Comment:
This is a complicated regex, so it should include comments to explain what each part is doing.

Also, the example from the comment should not be removed. If necessary, update that example or add a second one for the case this is dealing with.